gcc/config/riscv/riscv.c

```diff
@@ -4271,47 +4271,55 @@ riscv_gen_load_poly_int (rtx target, rtx tmp1, rtx tmp2, poly_int64 value)
   rtx insn;
 
   HOST_WIDE_INT scalar_offset = 0;
+  bool tail = true;
 
   if (value.coeffs[0] != value.coeffs[1])
     scalar_offset = value.coeffs[0] - value.coeffs[1];
 
   if (scalar_offset)
     value -= scalar_offset;
 
-  if (maybe_lt (value, BYTES_PER_RVV_VECTOR))
+  if (!maybe_lt (known_lt (value, 0) ? -value : value, UNITS_PER_V_REG))
```

Comment:
Stack value is negative, so we need to check with zero,
and then compared with UNITS_PER_V_REG